### PR TITLE
React Router rc1 hot loader fix

### PR DESCRIPTION
**Pull request by Faolain**
Great work man, I saw from the gitter chat that a few people were having trouble with the devhotloader breaking with the update to the new router. I hacked around with v1.0.0(rc1) release candidate and it seems to be all hooked up, albeit some warnings here and there.

### CODE REVIEW

**Comment by choonkending** [comment]
@Faolain Excellent work!

**Comment by joshhornby** [comment]
Might be worth knowing: 

https://twitter.com/dan_abramov/status/645595855475613697

**Comment by choonkending** [comment]
Thanks josh! On an unrelated note, I'm really sold on redux. You reckon
should create another repo for a boilerplate?

File: package.json [hunk 1]
```diff
@@ -59,11 +60,12 @@
     "passport-google-oauth": "^0.2.0",
     "passport-local": "^1.0.0",
     "react": "^0.13.3",
+    "react-router" : "^1.0.0-beta3",
     "react-helmet": "^1.0.1",
-    "react-router": "1.0.0-beta3",
```
Comment (choonkending) [inline package.json]: @Faolain Would you mind changing this to rc1 and try fixing transitionTo errors? Or would you prefer the last beta merged first and then a separate PR for rc1?

I've tested it with rc1 apart from the the onEnter errors where it checks if the user is logged in, everything works well. Good work with the `match` change in server.jsx. Thanks a lot man!

Comment (Faolain) [inline package.json]: Just updated the onEnter transition. No thanks needed, your kit has made my life and I'm sure the lives of others much easier.

**Comment by choonkending** [comment]
@Faolain good stuff! Tested to be working. Merging now.

**Comment by joshhornby** [comment]
@choonkending I reckon its best speaking to @gaearon because I think there are quite a few boilerplates already out.

**Pull request merged**